### PR TITLE
StateTimeline: Fix bug with state labels using stale y position cache during panel resize

### PR DESCRIPTION
**Pull request by drew08t**
Fixes https://github.com/grafana/support-escalations/issues/10649

this gets rid of the position cache and recalculates offsets during rendering. the cache was likely there for optimizing cursor/hover interactions, but it isnt being used for this any more.

Before:
Jun-05-2024 18-14-12

After:
Jun-05-2024 18-14-43

### CODE REVIEW

File: public/app/core/components/TimelineChart/TimelineChart.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export interface TimelineProps extends Omit<GraphNGProps, 'prepConfig' | 'propsT
   tooltip?: VizTooltipOptions;
 }
 
-const propsToDiff = ['rowHeight', 'colWidth', 'showValue', 'mergeValues', 'alignValue', 'tooltip'];
+const propsToDiff = ['rowHeight', 'colWidth', 'showValue', 'mergeValues', 'alignValue', 'tooltip', 'height'];
```
Comment (leeoniya) [inline public/app/core/components/TimelineChart/TimelineChart.tsx:24]: this will re-init the chart for each pixel that the chart is resized :(

Comment (drew08t) [reply public/app/core/components/TimelineChart/TimelineChart.tsx:24]: Maybe we can look for significant changes, defined as a function of row height.

**Comment by drew08t** [comment]
Thanks for the assist, @leeoniya! Your approach seems to have fixed it. I can no longer replicate the issue.

**Review by leeoniya — APPROVED** [review]
:shipit:

**Pull request merged**